src/System.Private.ServiceModel/tests/Scenarios/SelfHostWcfService/WcfService.cs

```diff
@@ -234,5 +236,46 @@ public XmlCompositeType GetDataUsingXmlSerializer(XmlCompositeType composite)
             composite.StringValue = composite.StringValue;
             return composite;
         }
+
+        public Stream GetStream(string data)
+        {
+            return StringToStream(data);
+        }
+
+        public bool UploadStream(Stream stream)
+        {
+            string data = StreamToString(stream);
+
+            if (!string.IsNullOrWhiteSpace(data))
+            {
+                return true;
```

Comment:
To respond to an earlier comment -- the server should *never* participate in the testing, such as verifying it got some magic content.  Server side code should instead be written to return something to the client so the test can verify what the server received was correct.  The operation can be written to control what the server does with its inputs, but the server should never directly know anything about how the calling test behaves.